// 加解密h定义
enum EncryptKeyDef {
	EncryptKeyLen = 32; // 加密密钥长度
	EncryptHeadLen = 2; // 加解密头大小，保存使用的密钥偏移量和使用的密钥字节数
	EncryptTailLen = 4; // 加解密尾大小，保存检验和

	AuthKeyLen = 16; // 认证串长度
	PublicKeyLen = 8; // 公钥长度
	PrivateKeyLen = 20; // 私钥长度
}

// 认证结果
enum AuthResult
{
	AUTH_OK = 0;
	AUTH_FAIL = 1;
}

// 登录请求
message LoginReq
{
	enum LoginType {
		NewRole = 0; // 新创建角色
		OldRole = 1; // 旧角色
	}

	required int32 clientVersion = 1; // 客户端版本号
    required string deviceid = 2; // 设备号
    required string notifyid = 3; // 通知号
    required string username = 4; // 用户名
	required int32 zoneid = 5; // 区ID
	required int32 userid = 6; //用户Id
	required int32 playerid = 7; // 玩家Id
	required string job = 8; // 职业
	required LoginType loginType = 9; // 操作类别
	required int32 authType = 10; //认证类型
	required string authKey = 11; //认证字符串
}

// 加解密密钥通知
message EncryptKeyNtf
{
	required string publicKey = 1; // 公钥，用于客户端进行加密
	required string privateKey = 2; // 私钥，用于客户端进行解密

	required string authKey = 3; // 认证串，已经过加密，需要客户端解出
}

// 身份验证请求
message AuthReq
{
	required string authKey = 1; // 认证串，已经过加密，需要服务端解出
}

// 身份验证答复
message AuthAck
{
	required AuthResult result = 1; // 认证结果[成功/失败]
	required string encryptKey = 2; // 密钥，此后的消息包均用此密钥加解密
}

// pingpong测试包
message PingPong
{
	required string pingpong = 1;
}